libglusterfs/src/inode.c

```diff
@@ -541,6 +539,7 @@ __inode_unref(inode_t *inode, bool clear)
     GF_ASSERT(inode->ref);
 
     --inode->ref;
+    this = THIS;
 
     index = __inode_get_xl_index(inode, this);
     if (index >= 0) {
```

Comment:
Do we really need to do this ? it could be useful for debugging in some cases, though I've never used it, but requiring `THIS` always just for a debugging purpose seems too costly.

Reply:
IIRC, we had a discussion about this in some other patches as well. TBH, I do not know what is the right step to take here. Maybe you and @mykaul can decide on if it is really needed else we can always drop this.

Reply:
@mohit84 @amarts @pranithk do you think it's useful/necessary to keep `xl_key` and `ref` into the ctx array ? IMO we could remove them or at least only keep them for debug builds

Reply:
Yes, I am fine with keeping them only in DEBUG builds. Production builds needn't have it IMO.